### PR TITLE
Carli/1223 zfp compression artifacts

### LINKED ISSUE #1223: weird 4x4 patterns in raster image due to zfp compression
The default zfp compression level is 11 and usually it works fine without noticeable defects. However, in some cases (I am guessing it is related to the distribution of NaN pixels which are common in the moment 1 and 2 images (velocity field and velocity dispersion) defects are visible such as the following example which is a moment image (velocity field). 

compression=11
NGC_4321_sci spw27 cube I pbcor 13CO_3-2 image moment weighted_coord-image-2020-10-8-10-28-48

compression=32
NGC_4321_sci spw27 cube I pbcor 13CO_3-2 image moment weighted_coord-image-2020-10-8-10-27-45

@kswang1029 should perform more experiments to see if we need a refined default compression level.

### PR DESCRIPTION
**Pull request by crocka**
**Description**

This PR fixes #1223 by increase the default compression ratio to 32 when the header unit of the image is km/s.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] changelog updated / ~~no changelog update needed~~
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
looks good. Tested with other moment1 (velocity field) or moment2 (velocity dispersion) images and all are looking fine. no regression from e2e tests.

File: src/stores/AppStore/AppStore.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ interface ViewUpdate {
     channel: number;
     stokes: number;
     focusPoint: Point2D;
+    headerUnit?: string;
```
Comment (YuHsuan-Hwang) [inline src/stores/AppStore/AppStore.ts]: Shall this be required instead of optional?

File: src/stores/AppStore/AppStore.ts [hunk 2]
```diff
@@ -1523,7 +1524,14 @@ export class AppStore {
                 // TODO: dynamic tile size
                 const tileSizeFullRes = reqView.mip * 256;
                 const midPointTileCoords = {x: midPointImageCoords.x / tileSizeFullRes - 0.5, y: midPointImageCoords.y / tileSizeFullRes - 0.5};
-                this.tileService.requestTiles(tiles, frame.frameInfo.fileId, frame.channel, frame.stokes, midPointTileCoords, this.preferenceStore.imageCompressionQuality, true);
+
+                // If BUNIT = km/s, adopted compressionQuality is set to 32 regardless the preferences setup
+                const bunitVariant = ["km/s", "km s-1", "km s^-1", "km.s-1"];
+                let compressionQuality = this.preferenceStore.imageCompressionQuality;
```
Comment (YuHsuan-Hwang) [inline src/stores/AppStore/AppStore.ts]: Perhaps simplify to one line as in `updateView`.

File: src/stores/AppStore/AppStore.ts [hunk 3]
```diff
@@ -1558,20 +1566,23 @@ export class AppStore {
...
         for (const update of updates) {
-            this.updateView(update.tiles, update.fileId, update.channel, update.stokes, update.focusPoint);
+            this.updateView(update.tiles, update.fileId, update.channel, update.stokes, update.focusPoint, update.headerUnit);
         }
     };
 
-    private updateView = (tiles: TileCoordinate[], fileId: number, channel: number, stokes: number, focusPoint: Point2D) => {
+    private updateView = (tiles: TileCoordinate[], fileId: number, channel: number, stokes: number, focusPoint: Point2D, headerUnit: string) => {
         const isAnimating = this.animatorStore.serverAnimationActive;
+        // If BUNIT = km/s, adopted compressionQuality is set to 32 regardless the preferences setup
```
Comment (YuHsuan-Hwang) [inline src/stores/AppStore/AppStore.ts]: Can be moved to `else` section.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes look good. 👍 Some minor comments:

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
@crocka Please update the change log, thanks.

**Pull request merged**